sdk/java/pulumi/src/main/java/com/pulumi/test/TestResult.java

```diff
@@ -0,0 +1,114 @@
+package com.pulumi.test;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.pulumi.core.Output;
+import com.pulumi.exceptions.RunException;
+import com.pulumi.resources.Resource;
+
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.stream.Collectors;
+
+/**
+ * A test result information.
+ */
+public class TestResult {
+    private final int exitCode;
+    private final List<Resource> resources;
+    private final List<Exception> exceptions;
+    private final List<String> errors;
```

Comment:
What's the difference between errors and exceptions?

Reply:
errors are error messages from logger, exceptions are exception instances

Reply:
we terminate on a logged error even if there were no exceptions (behaviour inherited from C# impl)

Reply:
Gotcha, so this basically captures what was logged.